feat: uffizzi integration

daramayis opened the pull request:
<!-- PR Checklist -->

# Description
Uffizzi CI/CD integration.

TattdCodeMonkey committed:
Merge pull request #588 from dcos-labs/nat/new-icons

Add new icons

vacas5 committed:
chore: broken build step (#597)

attempt to correct the broken builds by using npx

dependabot[bot] committed:
chore(deps): bump nested-object-assign from 1.0.3 to 1.0.4 (#582)

Bumps [nested-object-assign](https://github.com/Geta/NestedObjectAssign) from 1.0.3 to 1.0.4.
- [Release notes](https://github.com/Geta/NestedObjectAssign/releases)
- [Commits](https://github.com/Geta/NestedObjectAssign/compare/v1.0.3...v1.0.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump ini from 1.3.5 to 1.3.8 (#584)

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.8.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.8)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump elliptic from 6.5.3 to 6.5.4 (#589)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.3 to 6.5.4.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.3...v6.5.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump ua-parser-js from 0.7.22 to 0.7.28 (#590)

Bumps [ua-parser-js](https://github.com/faisalman/ua-parser-js) from 0.7.22 to 0.7.28.
- [Release notes](https://github.com/faisalman/ua-parser-js/releases)
- [Commits](https://github.com/faisalman/ua-parser-js/compare/0.7.22...0.7.28)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump merge-deep from 3.0.2 to 3.0.3 (#594)

Bumps [merge-deep](https://github.com/jonschlinkert/merge-deep) from 3.0.2 to 3.0.3.
- [Release notes](https://github.com/jonschlinkert/merge-deep/releases)
- [Commits](https://github.com/jonschlinkert/merge-deep/compare/3.0.2...3.0.3)

---
updated-dependencies:
- dependency-name: merge-deep
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump handlebars from 4.5.3 to 4.7.7 (#591)

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.5.3 to 4.7.7.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/handlebars-lang/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.5.3...v4.7.7)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump url-parse from 1.4.7 to 1.5.1 (#592)

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.4.7 to 1.5.1.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.4.7...1.5.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump hosted-git-info from 2.7.1 to 2.8.9 (#593)

Bumps [hosted-git-info](https://github.com/npm/hosted-git-info) from 2.7.1 to 2.8.9.
- [Release notes](https://github.com/npm/hosted-git-info/releases)
- [Changelog](https://github.com/npm/hosted-git-info/blob/v2.8.9/CHANGELOG.md)
- [Commits](https://github.com/npm/hosted-git-info/compare/v2.7.1...v2.8.9)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vacas5 committed:
chore: update semantic-release version (#598)

Upgrades semantic-release by two major versions, thus had to change the
releaserc config based on breaking changes. We'll also need to make sure
we're using LTS node version but Rodney and I couldn't come to a
decision about whether to introduce a node-version or nvmrc file.

dependabot[bot] committed:
chore(deps): bump postcss from 7.0.35 to 7.0.36 (#599)

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vacas5 committed:
fix: errors from react-toggled

This dependency was producing console errors in development. Deprecate
its usage but maintain parity by using state within the expandable
component.

vacas5 committed:
use newer convention for checking undefined, cx conditional convention

mstruebing committed:
Merge pull request #600 from dcos-labs/rja/fix/remove-react-toggled

fix: errors from react-toggled

mstruebing committed:
fix: use `display: none` to hide closed expendable items

mstruebing committed:
docs: add comment which component contains `willReceiveProps`

mstruebing committed:
Merge pull request #601 from dcos-labs/mstruebing/fix/D2IQ-65002/display-none

fix: use `display: none` to hide closed expendable items

vacas5 committed:
feat: title attr on truncated Text (#602)

When a user sends the wrap prop with the arg of truncate, in the case
that the child is a string, output the child as the title HTML attribute
so that the user and screen readers will get the string in its entirety.

Closes D2IQ-77202

clintonmedbery committed:
feat: add new locked config icons

Closes D2IQ-77204

clintonmedbery committed:
Merge pull request #603 from dcos-labs/clintonmedbery/feat/D2IQ-77204-airgapped-icons

feat: add new airgapped icons

clintonmedbery committed:
fix: updated lock config icon to be in system icons

Closes D2IQ-77204

clintonmedbery committed:
Merge pull request #604 from dcos-labs/clintonmedbery/feat/D2IQ-77204-airgapped-icons

fix: updated lock config icon to be in system icons

dependabot[bot] committed:
chore(deps): bump tar from 4.4.8 to 4.4.15 (#605)

Bumps [tar](https://github.com/npm/node-tar) from 4.4.8 to 4.4.15.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.8...v4.4.15)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vacas5 committed:
feat: new promo banner background gradient (#606)

* feat: new promo banner background gradient

Introducing a new oceanBlue version of the background gradient for the
PromoBanner, which has a specific implementation with three stops.

Closes D2IQ-77630

* tweak second stop per design

dependabot[bot] committed:
chore(deps): bump path-parse from 1.0.6 to 1.0.7 (#607)

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.6 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vacas5 committed:
fix: controlled FieldList row deletion (#609)

It seemed as though the onRemoveItem was never actually being called
based on the way that it was written, because the useEffect would not
fire, thus you'd end up with rendering issues. By tweaking a bit how
this function is called in the context, it appears to fix the issue.

I don't know why this callback function was written with the additional
closure, but I preserved it to maintain backwards compatibility.

Addresses D2IQ-77909

nataliepina committed:
chore: storybook v6 prep and theming

Closes D2IQ-77946

nataliepina committed:
refactor: migrate badge stories to csf
converted badge related components to functional components

nataliepina committed:
feat: added config-cycle icon

Closes D2IQ-77455

nataliepina committed:
chore: update storybook theming

nataliepina committed:
Merge pull request #612 from dcos-labs/npina/chore/update-theming

chore: update storybook theming

nataliepina committed:
Merge branch 'master' into npina/feat/D2IQ-77455

nataliepina committed:
Merge pull request #613 from dcos-labs/npina/feat/D2IQ-77455

D2IQ-77455 feat: added config-cycle icon

dependabot[bot] committed:
chore(deps): bump tar from 4.4.15 to 4.4.19

Bumps [tar](https://github.com/npm/node-tar) from 4.4.15 to 4.4.19.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.15...v4.4.19)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
chore: parallelize jenkins

try to run the validation steps in parallel

Commit:
chore: remove build storybook stage

nataliepina committed:
chore: migrate stories to csf

Closes D2IQ-78564

nataliepina committed:
Merge pull request #616 from dcos-labs/npina/chore/D2IQ-78564

Migrate Stories to CSF

nataliepina committed:
chore: update storybook to the latest version

Closes D2IQ-78642

nataliepina committed:
Merge pull request #617 from dcos-labs/npina/chore/D2IQ-78642

Update Storybook to the latest version

nataliepina committed:
feat: update to emotion 11 (#619)

BREAKING CHANGE: new emotion version and export theme exports changed

Closes D2IQ-73724

nataliepina committed:
feat: added breadcrumbitem component to handle (#621)

common formatting utilized inside of breadcrumb usage

Closes D2IQ-65928

dependabot[bot] committed:
chore(deps): bump tmpl from 1.0.4 to 1.0.5 (#620)

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Russell Anderson <randerson@d2iq.com>

nataliepina committed:
fix: update feature query effecting flex padding (#622)

brought out with emotion 11 update

nataliepina committed:
chore: update logo and set it to link as home (#625)

nataliepina committed:
chore: versioning upkeep storybook patch updates (#624)

nataliepina committed:
feat: add sticky section functionality to dropdown (#626)

Closes D2IQ-79942

nataliepina committed:
feat: update node and set version in dockerfile (#627)

nataliepina committed:
fix: fix for docker build (#628)

nataliepina committed:
fix: for console errors found in kommander (#629)

nataliepina committed:
chore: upgrade versions for devDependencies (#630)

nataliepina committed:
feat: add collection icon (#633)

Closes D2IQ-80931

nataliepina committed:
fix: update tabs and add story controls (#631)

BREAKING CHANGE
Before this fix tab spacing was set by default to 16px now it will be set to 24px.

Closes D2IQ-62145

nataliepina committed:
feat: add optional icon to dialogmodal (#632)

Closes D2IQ-80933

nataliepina committed:
fix: revert changes to dropdown menu (#634)

mstruebing committed:
feat: add optional `dataCy` attribute (#635)

This adds an optional `dataCy` attributes
to `ConfigurationMap{Label,Row,Value}` to be able
to search efficiently with cypress for that values

Related to D2IQ-81001

nataliepina committed:
fix: remove semantic-release-jira plugin (#636)

seialkali committed:
fix(package.json): remove node dependency from package.json (#638)

removing node from package.json because it shouldn't need to be listed as a dependency, and also it's causing issues when trying to run 'npm i' on an M1 Mac

nataliepina committed:
feat: set node/npm version in package.lock engines (#637)

nataliepina committed:
feat: improve grid features and alignment options (#639)

BREAKING CHANGE
Before this fix columnCount was required, new behavior includes defaults for when it is not set to become a fluid grid that sets the amount of rows and columns based on available space.
Before you could not center grid items, now the optional centerItems prop can be used.
Other features include a data-cy prop, and className for further customizations.

Closes D2IQ-81475

nataliepina committed:
fix: old npm version set in devDependencies (#641)

nataliepina committed:
fix: reorder addons to fix storybook warning (#642)

seialkali committed:
feat: update React version (#643)

To facilitate `npm link` usage we will need to update the version of React for ui-kit.
This can be aligned with the version in Kommander, which is currently v17.0.2.

BREAKING CHANGE: a new major version of React has been installed

Closes D2IQ-81484

nataliepina committed:
chore: update dependencies (#644)

Updating dependencies that do not require code changes.

Closes D2IQ-81764

nataliepina committed:
chore: update react related dependencies (#645)

nataliepina committed:
chore: update to storybook v6.4.3 (#646)

Closes D2IQ-81752

nataliepina committed:
chore: create github action for tests (#647)

nataliepina committed:
chore: update storybook with patch release (#648)

nataliepina committed:
chore: create github action for integration tests (#654)

nataliepina committed:
refactor: controls to graphic elements and charts (#651)

nataliepina committed:
chore: update github actions (#655)

nataliepina committed:
refactor: controls for navigation and layout (#653)

nataliepina committed:
refactor: controls for page structure and overlays (#650)

nataliepina committed:
refactor: controls for typography and style utils (#652)

nataliepina committed:
refactor: controls actions/feedback/data listing (#656)

nataliepina committed:
refactor: controls for utils and forms (#657)

nataliepina committed:
chore: remove knobs addon (#658)

Closes D2IQ-78643

nataliepina committed:
chore: github action for semantic release (#660)

seialkali committed:
refactor: typography controls updates (#664)

suggestions for improvements to typography controls
also includes the addition of a controlConstants.ts file

seialkali committed:
chore/refactor: swap TSLint for ESLint (#649)

* chore: swap tslint for eslint

TSlint is depreciated so we need to migrate over to ESlint
Have run tslint-to-eslint-config to convert TSlint rules into a ESlint config file
Then removed all TSlint packages and re-run npm i

Closes D2IQ-81753

* refactor: convert tslint to eslint

- fine tune eslint config
- remove tslint config files
- update readme
- convert tslint comments to eslint
- refactor code based on eslint errors where reasonable

* chore: add lint command to gh-actions

and remove lint and test command from Jenkins

nataliepina committed:
refactor: refactor args for stories (#665)

* refactor: refactor args for stories

* refactor: small tweaks

Add aspectRatio to AspectRatio story via args to autoselect specified option
Remove unused colors from colorcodedbadge story

Co-authored-by: Brittany LaGambina <blagambina@d2iq.com>

nataliepina committed:
Github Action - Deployment (#659)

* chore: github action for build and deploy

* ci: merge github actions into one job (#666)

Co-authored-by: Max Strübing <mxstrbng@gmail.com>

nataliepina committed:
feat: add insights product icon (#667)

Closes D2IQ-82103

nataliepina committed:
fix: loosen npm and node requirements (#668)

nataliepina committed:
fix: remove jenkins replaced by github actions (#663)

nataliepina committed:
feat: add fire icon (#669)

Closes D2IQ-82100

nataliepina committed:
chore: update to storybook v6.4.9 (#671)

Storybook suggested to install eslint-plugin-storybook also.

nataliepina committed:
feat: add card header for images (#670)

Closes D2IQ-82107

nataliepina committed:
Storybook Landing Page (#672)

* fix: update card styles

Fix for card styles when an image has a height and width included. Changes prevent the header image from overlapping card content, due to the previous height 100% for set all divs inside of the card.

* feat: welcome page and storybook setup for mdx

Added storybook updates, ability to import png as modules, images to use, card style updates, svgr for svgs, jest config for static assets, welcome page organization and related files.

Closes D2IQ-78978

nataliepina committed:
chore: sb version update (#673)

Update to safeguard against potential issues with faker.js and color.js which has affected storybook.

nataliepina committed:
docs: update docs to reference d2iq and copy edits (#674)

nataliepina committed:
chore: update cypress to latest v9 (#675)

nataliepina committed:
docs: add badgebutton story (#677)

nataliepina committed:
feat: management cluster workspace product icon (#678)

nataliepina committed:
chore: fixed versioning and added classnames (#676)

nataliepina committed:
fix: remove welcome components from packages (#679)

Update to tsconfig to apply typings to storybook doc component files.

Commit:
fix: unknown prop warnings (#680)

Add fixes for components which are throwing the unknown-prop warning.
https://reactjs.org/warnings/unknown-prop.html

Closes D2IQ-83375

nataliepina committed:
feat: allow text wrapping for tables (#681)

Closes D2IQ-84285

Commit:
fix: unknown prop warning - openInNewTab (#682)

nataliepina committed:
chore: update sb version and react types (#684)

nataliepina committed:
docs: add color documentation to storybook site (#683)

nataliepina committed:
docs: typography documentation (#687)

dependabot[bot] committed:
chore(deps): bump node-fetch from 2.6.6 to 2.6.7 (#686)

Bumps [node-fetch](https://github.com/node-fetch/node-fetch) from 2.6.6 to 2.6.7.
- [Release notes](https://github.com/node-fetch/node-fetch/releases)
- [Commits](https://github.com/node-fetch/node-fetch/compare/v2.6.6...v2.6.7)

---
updated-dependencies:
- dependency-name: node-fetch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Russell Anderson <randerson@d2iq.com>

dependabot[bot] committed:
chore(deps): bump shelljs from 0.8.4 to 0.8.5 (#685)

Bumps [shelljs](https://github.com/shelljs/shelljs) from 0.8.4 to 0.8.5.
- [Release notes](https://github.com/shelljs/shelljs/releases)
- [Changelog](https://github.com/shelljs/shelljs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/shelljs/shelljs/compare/v0.8.4...v0.8.5)

---
updated-dependencies:
- dependency-name: shelljs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nataliepina committed:
chore: update jest related dependencies (#688)

Commit:
feat: add thumbs up icon (#689)

nataliepina committed:
Prettier Update (#690)

* chore: update prettier deps

* refactor: prettier formatting

nataliepina committed:
refactor: updates for storybook site and readme (#692)

nataliepina committed:
fix: alignment issues in table header with icons (#691)

* fix: alignment issues in table header with icons

Fix for alignment issues seen with table headers that include the icon for tooltips.

* fix: table header cell wrapping issues

Closes D2IQ-85017

Commit:
fix: address Typescript errors in UIKitThemeProvider stories (#693)

Closes D2IQ-85007

nataliepina committed:
ci: include main in github actions (#695)

nataliepina committed:
Phase out master branch references (#696)

* docs: updating master references to main

* ci: phase out master and transition to main

nataliepina committed:
feat: new insights icon (#698)

nataliepina committed:
chore: storybook version update (#699)

nataliepina committed:
feat: fullscreenview banner component (#701)

dependabot[bot] committed:
chore(deps): bump minimist from 1.2.5 to 1.2.6 (#700)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mstruebing committed:
fix: uncontrolled to controlled component (#702)

Relates to D2IQ-83392

mstruebing committed:
fix: react-popper console errors (#703)

References D2IQ-87203

nataliepina committed:
refactor: remove sfc update prop types (#704)

Commit:
refactor: export FullscreenViewProps type (#706)

exporting this type so it can be used in kommander-ui for a custom component

nataliepina committed:
feat: add optional id and data-cy to props (#705)

Closes D2IQ-64874

Commit:
fix: change type for children prop in the Accordian component (#707)

The component was getting a Typescript error when used with only a single child, so
the type needed to be updated to allow for for both single and multiple children

nataliepina committed:
fix: tabindex values with updated versioning (#708)

nataliepina committed:
chore: emotion and storybook version updates (#709)

nataliepina committed:
chore: dep update maintenance (#710)

nataliepina committed:
fix: testing fix for react18 prop updates (#711)

nataliepina committed:
fix: badgebutton children and data-cy props (#712)

nataliepina committed:
ESLint Rule Review and Cleanup (#713)

* chore: update eslint related dependencies

* refactor: eslint rules and ignored files

* fix: required after eslint rule changes

Closes D2IQ-82132

clintonmedbery committed:
chore: update faker dependency (#714)

Closes D2IQ-88823

nataliepina committed:
feat: new default and custom colors for appchrome (#715)

These changes affect the AppChrome sidebar and header style.
Changes include refactoring of components such as removing React.FC.

BREAKING CHANGE: The fallback theme colors for the AppChrome component have been updated. The header fallback has changed from purple to white. The sidebar
fallback has changed from black to gray.

clintonmedbery committed:
chore: added a watch option for development (#716)

Commit:
chore: reintroduce multiselect typeahead story (#718)

mstruebing committed:
chore: update svgo (#719)

* Closes D2IQ-88825

nataliepina committed:
chore: storybook 6.5 upgrade (#720)

Commit:
fix: reintroduce dropped typeahead stories (#721)

During a refactor a few of these examples got dropped and stopped
functioning. Adding them back in to help with regression testing of the
downshift upgrade.

Additionally I discovered that the complex list items example doesn't
work even when adding back in args to the default. This apparently has
not worked since some time in 2019, as of the following commit https://github.com/dcos-labs/ui-kit/commit/1c01ea9bbbaf49047f5098fa80f0ab694f02d8fa

Thus, I removed it completely rather than trying to untangle it. We
evidently have not needed it.

Commit:
chore: upgrade downshift (#722)

not sure exactly how this test was meant to work prior to this but it
seems like the user behavior is what we want, it just skips over the
disabled item and allows you to select the first available.

Otherwise some type updates to get it to work. Could use some visual
regression testing on these components.

Closes D2IQ-88824

nataliepina committed:
feat: custom data-cy and classname props (#725)

clintonmedbery committed:
chore: removed max size deprecated dependency (#724)

This feature was not working and I am not sure replacing it is worth the effort.

Closes D2IQ-88827

Commit:
feat: add gap property to flex component (#726)

Adding the gap property to the Flex component so the gutter size for columns and rows can be set at the same time.

Closes D2IQ-90211

nataliepina committed:
chore: sb update with suggested changes (#728)

nataliepina committed:
feat: color prop for sectionloadingindicator (#727)

Ability to set a custom color for the SectionLoadingIndicator.
Loading indicators story improvements included.

dependabot[bot] committed:
chore(deps): bump npm from 8.8.0 to 8.12.0 (#729)

Bumps [npm](https://github.com/npm/cli) from 8.8.0 to 8.12.0.
- [Release notes](https://github.com/npm/cli/releases)
- [Changelog](https://github.com/npm/cli/blob/latest/CHANGELOG.md)
- [Commits](https://github.com/npm/cli/compare/v8.8.0...v8.12.0)

---
updated-dependencies:
- dependency-name: npm
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
refactor: fix pagination story (#731)

dependabot[bot] committed:
chore(deps): bump semver-regex from 3.1.3 to 3.1.4 (#732)

Bumps [semver-regex](https://github.com/sindresorhus/semver-regex) from 3.1.3 to 3.1.4.
- [Release notes](https://github.com/sindresorhus/semver-regex/releases)
- [Commits](https://github.com/sindresorhus/semver-regex/commits/v3.1.4)

---
updated-dependencies:
- dependency-name: semver-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
refactor: introduce chromatic (#730)

* refactor: introduce chromatic

Chromatic is a tool that will give us side by side visual comparisons of
our storybook with the one deployed on main, as well as ephemeral
environments with every branch that is pushed. This should help us QA
quickly and find more bugs.

* refactor: remove faker js dependency

Using faker to generate data on each render would negate much of the
benefit of Chromatic, which will automatically do screenshot diffs
between branches to make visual changes easier to glean and QA.

There is questionable benefit to faking this every time versus simply
using some static data for the rendering. I don't think this dependency
will be particularly missed.

nataliepina committed:
feat: a11y regression testing for storybook (#733)

Closes D2IQ-90239

dependabot[bot] committed:
chore(deps-dev): bump semantic-release from 19.0.2 to 19.0.3 (#734)

Bumps [semantic-release](https://github.com/semantic-release/semantic-release) from 19.0.2 to 19.0.3.
- [Release notes](https://github.com/semantic-release/semantic-release/releases)
- [Commits](https://github.com/semantic-release/semantic-release/compare/v19.0.2...v19.0.3)

---
updated-dependencies:
- dependency-name: semantic-release
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nataliepina committed:
fix: remove linechart and chartist deps (#736)

BREAKING CHANGE:
The LineChart component and it's chartist and react-chartist dependencies are removed.

nataliepina committed:
chore: misc dep updates (#735)

nataliepina committed:
chore: storybook 6.5.9 dep updates (#737)

nataliepina committed:
chore: update types and eslint related deps (#738)

mstruebing committed:
chore: add codeowners (#739)

* Closes D2IQ-90977

dependabot[bot] committed:
chore(deps): bump parse-url from 6.0.0 to 6.0.2 (#740)

Bumps [parse-url](https://github.com/IonicaBizau/parse-url) from 6.0.0 to 6.0.2.
- [Release notes](https://github.com/IonicaBizau/parse-url/releases)
- [Commits](https://github.com/IonicaBizau/parse-url/commits)

---
updated-dependencies:
- dependency-name: parse-url
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Russell Anderson <randerson@d2iq.com>

nataliepina committed:
chore: minor and patch dependency updates (#741)

mstruebing committed:
chore: update tablemark and switch module system (#742)

dependabot[bot] committed:
chore(deps): bump terser from 4.8.0 to 4.8.1 (#743)

Bumps [terser](https://github.com/terser/terser) from 4.8.0 to 4.8.1.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nataliepina committed:
refactor: welcome page to hide canvas tab (#744)

nataliepina committed:
refactor: avatar and icon types, props, and docs (#746)

nataliepina committed:
refactor: pull request template updates (#748)

mstruebing committed:
Rework Githooks & Update commitlint (#750)

* chore: update commitlint

* Closes D2IQ-90924

* chore: introduce husky and remove custom scripts

mstruebing committed:
chore: remove not needed deps (#751)

I initially wanted to update one dependency and found one unused dependency.
I decided to go over all deps and see which ones are unused.
I executed every script and copied the dist directory into kommander-ui
after rebuilding to see if everything still works. I did not notice anything wrong.

clintonmedbery committed:
refactor: avatar and icon types, props, and docs (#746) (#749)

Co-authored-by: Natalie Pina <34781875+nataliepina@users.noreply.github.com>

nataliepina committed:
chore: updates for patch and minor deps (#752)

nataliepina committed:
Remove Cypress Integration Tests (#754)

* ci: remove cypress integration tests from ci

* docs: remove cypress integration test info

* refactor: remove cypress integration testing

nataliepina committed:
chore: update storybook-test-runner dependency (#753)

nataliepina committed:
build: migrate ts-loader to babel-loader (#755)

* build: migrate ts-loader to babel-loader

Closes D2IQ-92433

* ci: add type checking step

Commit:
refactor: form controls to functions (#756)

Refactoring some form PureComponents to be functions instead. One thing
that tripped me up was that the refactor of SelectInput threw warnings
if appearance was a required prop. Not sure how this wasn't throwing
    warnings before. Either way though, because it is defaulted, making
    it an optional prop is still backwards compatible.

nataliepina committed:
refactor: indeterminate definition and prop typing (#757)

Closes D2IQ-92602

nataliepina committed:
chore: dependency maintenance sept 2022 (#758)

nataliepina committed:
chore: update concurrently (#760)

Included patch update for eslint-plugin-react.

mstruebing committed:
refactor: make onSelect and selectedIndex props optional (#762)

* let children be an array or single element

mstruebing committed:
fix: dont use default props for `selectedIndex` and `onSelect` (#763)

mstruebing committed:
fix: make `inputLabel` optional (#764)

mstruebing committed:
fix: let `dist` command output transpiled js (#765)

We have introduced type checks and therefore changed our
tsconfig file to only emit types to be faster.
Our dist tsconfig file extends this other file so no javascript files
were transpiled

Commit:
refactor: card components and breadcrumb (#766)

swapping out class components for functions and also removing as much
enzyme as I can. Using react-test-renderer where possible for snapshots.

clintonmedbery committed:
chore: update typescript (#767)

* chore: update typescript to latest

Closes D2IQ-90920

* chore: updated snapshots

clintonmedbery committed:
chore: jest update (#769)

* chore: updated jest

* chore: updated snapshots for jest 29

Commit:
refactor: checkbox helper (#768)

Co-authored-by: Max Strübing <mxstrbng@gmail.com>

Commit:
fix: revert module change (#774)

This steps back a complete shift from CommonJS to esm modules, and
instead makes only the file that imports tablemark use esm. Then, we use
a dynamic import statement inside the build script to import that file
as a workaround. Thus we're able to leave tablemark as is for now.

Closes D2IQ-92904

Commit:
refactor: hashmap and corresponding components (#772)

Refactored hashmap from a class to a function. Tried to make some type
improvements and also to refactor some of the tests. Had trouble using
react-test-renderer create with one of the tests that used a mock
function, possibly because of children further down the dom tree, so I
skipped it.

Commit:
fix: pagination as links bug, fullscreen props (#773)

* fix: pagination as links bug, fullscreen props

This began as a refactor of several components away from classes. While
doing so I was able to remove the Delegate library that was throwing
react errors because it was very out of date. Additonally I found a bug
with the way proptypes were defined for headerComponent in two
components, and also a bug with the pagination when trying to use the
pager buttons as links.

* refactor: allow any to preserve bwards compat

Commit:
refactor: story helpers to functions (#775)

I didn't get all the helpers but these are the least risky to refactor
because they don't affect the underlying components. Still a long way to
go but getting there.

Commit:
refactor: tooltip and pageheader (#777)

nataliepina committed:
refactor: link components (#745)

* refactor: link components

Converting link components to functional components with updated types and additional cleanup.

* refactor: uikitprovider to a function component

And UIKitThemeProvider

* refactor: focusstylemanager

Migrating FocusStyleManager to a functional component.

Commit:
refactor: typeahead component and stories (#776)

This one was more complicated but converted Typeahead from a class
component to a functional component. All functionality should be
preserved, but I did find that the CustomMenuEmptyState story did not
work as expected because the empty state should be shown when there are
no items. Also made a decision to remove the filteredlist typeahead
because it didn't really demonstrate any functionality of the component
itself, just of how a wrapping component could control its state. It
seemed superfluous.

clintonmedbery committed:
test: start to remove enzyme (#771)

* chore: updated jest

* chore: updated snapshots for jest 29

* test: added react testing library and updated accordion test

* test: updated avatar, sidebar and appchrome tests

* test: updated badge, color badge, icon and other tests

* test: updated other tests to use asFragment

* docs: updated information about snapshot tests

* test: added lint and test ypes to pre-commit

JohnRamberger committed:
refactor: infobox component (#780)

* refactor: infobox component

Converted the infobox component to a function component.

* refactor: infobox component

nataliepina committed:
ci: update code checks and formatting (#778)

TGlide committed:
Refactor ToggleBoxGroup Component (#789)

* refactor: change ToggleBoxGroup to fn component

* refactor: change ToggleBoxGrouPStoryHelper to fn component

* refactor: minor code style changes

* refactor: memoize ToggleBoxGroup

leonardomaier committed:
refactor: turn Textarea into a functional component (#795)

davidtaing committed:
refactor: convert FormFieldWrapper to FC (#791)

* refactor: convert FormFieldWrapper to FC

fixes #787

* perf: wrap FormFieldWrapper in React.Memo

& update Jest snapshot tests.

nataliepina committed:
refactor: migration away from react.fc (#797)

nataliepina committed:
refactor: update tabs docs and tabitem to fc (#794)

leonardomaier committed:
refactor: turn ToggleWrapper into a function component (#804)

chan2k14 committed:
refactor: DropdownContents Component to Function Component (#798)

* refactor: DropdownContents Component to Functional Component

* refactor(fixed lint errors): DropdownContents Component to Function Component

* refactor(fixed lint errors): DropdownContents Component to Function Component [2nd revision]

* refactor(fixed lint errors): DropdownContents Component to Function Component [3rd revision]

* refactor(added test snapshot): DropdownContents Component to Function Component

* feat{Dropdownable}: Replaced react-click-outside with react-detect-click-outside library and refactored components

* feat{fixed lint errors}: Replaced react-click-outside with react-detect-click-outside library and refactored components

* feat{fixed lint errors}: Replaced react-click-outside with react-detect-click-outside library and refactored components [2nd revision]

* feat{refactored DropdownStoryFit}: Replaced react-click-outside with react-detect-click-outside library and refactored components

* feat(added package-lock.json): Replaced react-click-outside with react-detect-click-outside library and refactored components

leonardomaier committed:
refactor: update Typeahead component tests from enzyme to react-testing-library (#806)

nataliepina committed:
refactor: tests from enzyme to react-testing-lib (#807)

vishnuvnathan committed:
Refactor/clickable to fc (#808)

* refactor: clickable component to functional component

* refactor: clickable component wrapped in memo

* test: updated the snapshots

* refactor: clickable functional component

Co-authored-by: ntpl lap 411 <vishnu.viswanathan@neoito.com>

dependabot[bot] committed:
chore(deps): bump minimist and commitizen (#812)

Bumps [minimist](https://github.com/minimistjs/minimist) and [commitizen](https://github.com/commitizen/cz-cli). These dependencies needed to be updated together.

Updates `minimist` from 1.2.5 to 1.2.6
- [Release notes](https://github.com/minimistjs/minimist/releases)
- [Changelog](https://github.com/minimistjs/minimist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/minimistjs/minimist/compare/v1.2.5...v1.2.6)

Updates `commitizen` from 4.2.4 to 4.2.5
- [Release notes](https://github.com/commitizen/cz-cli/releases)
- [Commits](https://github.com/commitizen/cz-cli/compare/v4.2.4...v4.2.5)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
- dependency-name: commitizen
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nataliepina committed:
docs: story for infobox component (#809)

Commit:
feat: add message control to infobox stories (#816)

davidtaing committed:
refactor: migrate DropdownMenu Tests to RTL (#814)

* refactor: cleanup tests in preparation

for migration to react-testing-library

Summary:
- Simplified logic
- Rename test for better clarity

* refactor: cleanup tests in preparation

for migration to react-testing-library

Summary:
- Remove focus and blur events as they had not effect on the test result
- Renamed test for better clarity

* refactor: cleanup tests in preparation

for migration to react-testing-library

Summary:
- Removed events that had no effect on test results
- Renamed test for better clarity

* refactor: cleanup tests in preparation

for migration to react-testing-library

Summary:
- Test still passed after removing all events.
- Added expect statement to make the test more robust
- Removed events that had no effect on test results

* refactor: migrate tests to react-testing-library

and update snapshots

* fix: typo

* refactor: rename tests for better conciseness

* test: rebuild failed snapshot tests

Tests cases were checked for false-positives by changing render text and
seeing if the tests failed.

clintonmedbery committed:
docs: added information about baseelement (#820)

leonardomaier committed:
refactor: update Popover component tests from enzyme to react-testing-library (#821)

Commit:
refactor: convert ConfigurationMap tests from enzyme to RTL (#822)

nataliepina committed:
refactor: more refactoring away enzyme to rtl (#815)

Migrating to react testing library

Commit:
feat: optional max-width for FullscreenView header (#824)

If child content of FullscreenView is wrapped in a container, it would
be nice to be able to contain the header of this component also for
visual consistency. Opted to keep the padding as is and just account for
it in the max width so that consumers can pass in our design tokens as
is without having to do the math themselves to account for the padding
left and right.

Addresses D2IQ-92266

nataliepina committed:
refactor: move wrapping react.memo to export (#827)

Commit:
fix: contain FullscreenView with classname (#828)

In order to support responsive layouts and different container sizes at
various widths, this really should just take a classname so that a
client has total control. I tried to borrow the container classname from
the container component and go with a boolean argument like "contained"
but I found that our usage in DKP needed to change the box-sizing anyway
to get it to be consistent. Rather than have both props, just went with
one, more versatile prop.

alberturria committed:
Refactor TextInput related components from classes to functional components (#801)

* refactor: textInput to functional component

* refactor: textInputWithIcon to functional component

* refactor: textInputWithButtons to functional component

* refactor: textInputWithBadges to functional component

* fix: remove iconStart from props

nataliepina committed:
refactor: togglecontent component, story, and test (#826)

* refactor: togglecontent component, story, and test

* refactor: file name casing

leonardomaier committed:
refactor: converts ToggleInputList class component into a function component (#829)

nataliepina committed:
refactor: remove statelesscomponent use (#837)

React.StatelessComponent is deprecated.

nataliepina committed:
refactor: togglebox story (#834)

Updates to improve the functionality of the ToggleBox story.
Converted ToggleBoxStoryHelper to a function component.

davidtaing committed:
refactor: DialogModal to Function Component (#831)

* refactor: migrate DialogModal to FC

Migrate DialogModal from Class Component to Function Component

* refactor: migrate DialogModalWithFooter to FC

Migrate DialogModalWithFooter from Class Component to Function Component

* refactor: migrate ModalBase to FC

Migrate ModalBase from Class Component to Function Component

* refactor: update failed snapshot tests

due to ModalBase's migration to Function Component from Class Component

* perf: wrapped refactored components in React.memo

- DialogModal
- DialogModalWithFooter
- ModalBase

* refactor: update failed snapshot tests

from newly refactored components being wrapped in React.memo
- DialogModal
- DialogModalWithFooter
- ModalBase

* refactor: add missing useEffect dep

so that useEffect fires when the isOpen prop changes.

* refactor: remove ModalExample.tsx

as it was an empty file.

nataliepina committed:
chore: dependency maintenance (#836)

nataliepina committed:
refactor: buttonbase from class to fc (#838)

Commit:
feat: uffizzi integration

Aksa12 committed:
refactor: migrate Clickable and CodeSnippet components tests to RTL (#844)

Co-authored-by: aksa <aksa@jobcase.com>

Aksa12 committed:
refactor: migrate Overlay from class to function component (#839)

Co-authored-by: aksa <aksa@jobcase.com>

derekhassan committed:
refactor: textinputwithbadgestypeahedstoryhelper from class to function component (#845)

taksuparth committed:
refactor: fieldList tests (#848)

taksuparth committed:
refactor: selectInput tests (#851)

taksuparth committed:
Refactor test cases of dropdownable and dropdown menu item (#849)

* refactor: dropdownable tests

* chore: deletes dropdownMenuItem tests

axisofentropy commented on docker-compose.uffizzi.yml:
```diff
@@ -0,0 +1,19 @@
+    deploy:
+      resources:
+        limits:
+          memory: 4000m
```
I don't think the `m` notation makes sense here. Try the default `125M` and work upwards from there.  More info here https://docs.uffizzi.com/references/compose-spec/#resources-limits-memory

Also Uffizzi ignores `resources.cpu` specifications.

daramayis replied:
- `resources.cpu` has been removed.
- Tried following memory values
  - `125M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3318915889
  -  `500M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3319058470/jobs/5483736212
  - `1000M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3319278317
  - `2000M` - app preview deployed successfully in 8 minutes.

axisofentropy commented on docker-compose.uffizzi.yml:
```diff
@@ -0,0 +1,19 @@
+    image: "${APP_IMAGE}"
+    ports:
+      - "6006:6006"
+    restart: always
```
I believe Uffizzi ignores this `restart` specification. But Kubernetes does always try to restart containers.

daramayis replied:
OK, I will remove it.

nataliepina committed:
docs: comments to clarify modifierUtils.ts file (#858)

taksuparth committed:
refactor: convert enzym tests to react-testing-library-tests (#847)

* refactor: promos tests

* refactor: pagination tests

* refactor: segmentedControl tests

taksuparth committed:
refactor: tooltip tests (#852)

* refactor: tooltip tests

* chore: remove unused snapshot

Commit:
fix: forwardrefs to FocusStyleManager children (#843)

I am not sure that I did this correctly. This is a tough then where the
child component could literally be anything, so typing the ref is very
difficult. In some cases its a button, others an a, sometimes a label.
In any event, for the ui-kit components that are children of
FocusStyleManager, I wrapped them in forwardRef. Resolves the console
warning issues for the pages that use FocusStyleManager.

Closes D2IQ-93522

manav-06 committed:
refactor: migrate table component tests from enzyme to rtl (#846)

* refactor: migrate table component tests from enzyme to rtl

* refactor: using testid to test Sorting

* refactor: updated onstatechange test

* refactor: destructure getByTestId from render

taksuparth committed:
refactor: textInput tests (#857)

* refactor: textInput tests

* refactor: textInputWithButtons tests

* refactor: textArea tests

* refactor: textInputWithBadges tests

taksuparth committed:
refactor: toggleInputList tests (#859)

* refactor: toggleInputList tests

* refactor: toggleBoxGroup tests

Commit:
fix: rollback ref on unstyled link (#864)

* fix: rollback ref on unstyled link

Consuming LinkComponent has been a challenge downstream. We don't
actually need the focus style manager to handle unstyled links as far as
I can tell, so the console warnings should be resolved without having to
keep drilling down into child Link components.

* test: update pagination snap

Commit:
fix: console error for togglewrapper (#865)

React does not allow checked prop on uncontrolled components. Need to
use defaultChecked. This quiets the consoles showing up when running
tests.

taksuparth committed:
fix: make sortable table header chevron clickable (#860)

* fix: snapshots

* fix: make sortable table header chevron clickable

* chore: remove not required changes

* chore: resolve conflicts

* chore: resolve comments

* chore: updates screenshot

Commit:
test: clean up toast test suite (#866)

* test: clean up toast test suite

The toast test suite was tightly coupled with Toaster. This may have
been a copy paste issue or something, but the snapshots didn't even
reflect the markup of the Toast component. I have removed the wrapping
of each of these in Toaster, and a couple of tests that were targeting
behavior of Toaster rather than Toast. I have refactored to use RTL
instead of enzyme in other cases.

* refactor: toast to function

nataliepina committed:
refactor: tests to rtl (#868)

Refactored the TextInputWithIcon and Modal tests

Commit:
fix: toaster stories (#869)

During a separate refactor I found a bug in how I did the stories for
Toaster. Autodismiss should dismiss one at a time rather than all at
once. I think this is a bit more conventional react anyway. Also added a
story for multiple toasts at the same time.

axisofentropy commented on .github/workflows/uffizzi-build.yml:
```diff
@@ -0,0 +1,86 @@
+
+jobs:
+
+  build-appication:
```
typo

daramayis replied:
Fixed.

axisofentropy commented on .github/workflows/uffizzi-build.yml:
```diff
@@ -0,0 +1,86 @@
+        uses: actions/checkout@v3
+      - name: Generate UUID image name
+        id: uuid
+        run: echo "::set-output name=uuid::$(uuidgen)"
```
We're trying to deprecate `set-output`. the sooner the better.  Only change it here if this PR is likely to be accepted.  This is a good example https://github.com/UffizziCloud/quickstart/commit/3731ddb33771ddda88d93db2356dddf48f1ae7b6

We can also upgrade the third-party actions we're using like in this commit https://github.com/UffizziCloud/quickstart/commit/af17b37b9513f28e6299577611947ec8fe3cdade

daramayis replied:
Done.

daramayis replied:
As we agreed - we will apply this change later.

axisofentropy commented on .github/workflows/uffizzi-build.yml:
```diff
@@ -0,0 +1,86 @@
+        run:  |
+          cat << EOF > event.json
+          ${{ toJSON(github.event) }} 
+          
```
empty line could get into JSON file

daramayis replied:
Done.

daramayis replied:
As we agreed - we will apply this change later.

axisofentropy commented on .github/workflows/uffizzi-build.yml:
```diff
@@ -0,0 +1,86 @@
+    steps:
+      # If this PR is closing, we will not render a compose file nor pass it to the next workflow.
+      - name: Serialize PR Event to File
+        run: echo '${{ toJSON(github.event) }}' > event.json
```
For quote safety this should be implemented as above on line `61`.

daramayis replied:
Done.

daramayis replied:
As we agreed - we will apply this change later.

axisofentropy commented on docker-compose.uffizzi.yml:
```diff
@@ -0,0 +1,17 @@
+services:
+  application:
+    image: "${APP_IMAGE}"
+    ports:
```
I don't think this does anything on Uffizzi, but it may be helpful for people reading this file. Optional.

daramayis replied:
Yeah, it's helpful when we read this file :)

axisofentropy commented:
I added some comments but they're mostly optional changes. This PR looks good!

axisofentropy commented on .github/workflows/uffizzi-preview.yml:
```diff
@@ -0,0 +1,83 @@
+    name: Use Remote Workflow to Preview on Uffizzi
+    needs:
+      - cache-compose-file
+    uses: UffizziCloud/preview-action/.github/workflows/reusable.yaml@v2
```
Fix to a specific version of this reusable workflow so we don't accidentally break this in the future. Right now that's `v2.6.0`.

daramayis replied:
Done

daramayis replied:
Done.

taksuparth committed:
Refactor testcases expandable (#863)

* refactor: exapndable testcases

* chore: address comments

Commit:
test: toaster tests with RTL (#867)

* test: toaster tests with RTL

Some of these tests were testing internal component state. Opted instead
to try and use visual cues that would result from such a state. This
will make refactoring the component much easier.

* test: refactor toaster class function test

This rewrites a test to test the user facing behavior rather than
testing that certain class functions are being called. Paves the way for
a functional refactor of this component.

* refactor: toaster component to function

* refactor: small things to fix transitions

nataliepina committed:
chore: remove enzyme dependencies and references (#870)

nataliepina committed:
fix: refactor toggleboxgroup to prevent rerenders (#872)

nataliepina committed:
feat: custom classname and data-cy props (#873)

dependabot[bot] committed:
chore(deps): bump loader-utils from 1.4.0 to 1.4.1 (#874)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.0 to 1.4.1.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.1/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.0...v1.4.1)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nataliepina committed:
chore: node suggested engines (#871)

nataliepina committed:
feat: react 18 (#875)

includes test and type updates to support react 18

Closes D2IQ-92806

nataliepina committed:
Dependency Updates and MDX2 support (#876)

* chore: update dependencies

* fix: resolve mdx2

nataliepina committed:
docs: steps and check for downstream testing (#877)

Added a PR template check for testing downstream changes.
Updated the documentation around pre-release testing with npm-link.

nataliepina committed:
fix: sorter chevron alignment in table header (#879)

The chevron is bumped down below the header title when a header tooltip is present.

Closes D2IQ-94167

dependabot[bot] committed:
chore(deps): bump loader-utils from 1.4.1 to 1.4.2 (#880)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.1 to 1.4.2.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.2/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.1...v1.4.2)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
